muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatSyncController.java

```diff
@@ -163,39 +157,157 @@ public void syncStudent(SchoolDataIdentifier studentIdentifier){
         }
       }
     } catch (Exception e) {
       logger.log(Level.INFO, "Exception when syncing user " + studentIdentifier.getIdentifier(), e);
     } 
   }
+  
+  public void removeChatRoomMembership(SchoolDataIdentifier studentIdentifier, WorkspaceEntity workspaceEntity) {
+    
+    String openfireToken = pluginSettingsController.getPluginSetting("chat", "openfireToken");
+    if (openfireToken == null) {
+      logger.log(Level.INFO, "No openfire token set, skipping room sync");
+      return;
+    }
+
+    String openfireUrl = pluginSettingsController.getPluginSetting("chat", "openfireUrl");
+    if (openfireUrl == null) {
+      logger.log(Level.INFO, "No openfire url set, skipping room sync");
+      return;
+    }
+
+    String openfirePort = pluginSettingsController.getPluginSetting("chat", "openfirePort");
+    if (openfirePort == null) {
+      logger.log(Level.INFO, "No openfire port set, skipping room sync");
+      return;
+    }
+    if (!StringUtils.isNumeric(openfirePort)) {
+      logger.log(Level.WARNING, "Invalid openfire port, skipping room sync");
+      return;
+    }
+
+    AuthenticationToken token = new AuthenticationToken(openfireToken);
+    RestApiClient client = new RestApiClient(openfireUrl, Integer.parseInt(openfirePort, 10), token);
+    
+    User user = userController.findUserByDataSourceAndIdentifier(studentIdentifier.getDataSource(), studentIdentifier.getIdentifier()); 
+
+    String userSchoolDataSource = user.getSchoolDataSource();
+    String userIdentifier = user.getIdentifier();
+    
+    Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
+
+    String roomName = workspace.getIdentifier();
+    String jid = userSchoolDataSource + "-" + userIdentifier;
+    
+    client.deleteMember(roomName, jid);
+  }
+  
+  public void removeWorkspaceChatRoom(WorkspaceEntity workspaceEntity) {
+    
+    String openfireToken = pluginSettingsController.getPluginSetting("chat", "openfireToken");
+    if (openfireToken == null) {
+      logger.log(Level.INFO, "No openfire token set, skipping room sync");
+      return;
+    }
+
+    String openfireUrl = pluginSettingsController.getPluginSetting("chat", "openfireUrl");
+    if (openfireUrl == null) {
+      logger.log(Level.INFO, "No openfire url set, skipping room sync");
+      return;
+    }
+
+    String openfirePort = pluginSettingsController.getPluginSetting("chat", "openfirePort");
+    if (openfirePort == null) {
+      logger.log(Level.INFO, "No openfire port set, skipping room sync");
+      return;
+    }
+    if (!StringUtils.isNumeric(openfirePort)) {
+      logger.log(Level.WARNING, "Invalid openfire port, skipping room sync");
+      return;
+    }
+
+    AuthenticationToken token = new AuthenticationToken(openfireToken);
+    RestApiClient client = new RestApiClient(openfireUrl, Integer.parseInt(openfirePort, 10), token);
+    
+    
+    Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
```

Comment:
I'm fairly certain the identifier in workspaceEntity (Muikku's representation of a course) is the exact same as the identifier in Workspace (Pyramus' representation of a course) so instead of...

`Workspace workspace = workspaceController.findWorkspace(workspaceEntity);`
`String roomName = workspace.getIdentifier();`
`client.deleteChatRoom(roomName);`

...you could probably do with just...

`client.deleteChatRoom(workspaceEntity.getIdentifier());`

...so there would be one less server call to Pyramus.